File: Java&Kotlin lecture/src/main/java/com/java/student/GraduatedStudent.java
package com.java.student;

public class GraduatedStudent extends Student{
    int thesis;
    static int PASS = 70;

    public GraduatedStudent(String id, String name, int english, int math, int thesis) {
        super(id, name, english, math);
        this.thesis = thesis;
    }

    @Override
    public void showScore() {
        int average = getAverage();
        System.out.println(name + "\tEnglish:" + english + "\tMath:" + math + "\tThesis:" + thesis);
        System.out.print(average > PASS ? "PASS" : "FAILED");

        char grading = 'F';

        switch (average / 10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'E';
                break;
        }

        System.out.println("\t" + grading);
    }

    @Override
    public int getHighest() {
        if(english >= math && english >= thesis) {
            System.out.print("English ");
            return english;
        }else if(math >= english && math >= thesis) {
            System.out.print("Math ");
            return math;
        } else {
            return thesis;
        }
    }

    @Override
    public int getAverage() {
        return (english + math + thesis) / 3;
    }
}

File: Java&Kotlin lecture/src/main/kotlin/com/kotlin/student/GraduatedStudent.kt
package com.kotlin.student

class GraduatedStudent constructor(id: String, name: String? , english: Int, math: Int, var thesis: Int) :
    Student(id, name, english, math) {
    companion object {
        const val PASS = 70
    }

    override fun showScore() {
        val average = getAverage()
        println("$name\tEnglish:$english\tMath:$math\tThesis:$thesis")
        println(if(average > PASS) "PASS" else "FAILED")

        val grading: Char = when (average) {
            in 90 .. 100 -> 'A'
            in 80 .. 89 -> 'B'
            in 70 .. 79 -> 'C'
            in 60 .. 69 -> 'D'
            else -> 'E'
        }

        println("\t" + grading)
    }

    override fun getHighest(): Int {
        return if(english >= math && english >= thesis){
            print("English ")
            english
        }else if(math >= english && math >= thesis){
            print("Math ")
            math
        } else {
            print("Thesis ")
            thesis
        }
    }

    override fun getAverage(): Int = (english + math + thesis) / 3
}